### ISSUE TITLE
ValidationWebhookConfiguration keeps getting updated

### ISSUE THREAD

**Issue by style95** [istio/istio#52741]
I am testing how many EnvoyFilters can be created.
I had created around 200k EnvoyFilters and kept them for a couple of days.
Right after creating all EnvoyFilters, it looked fine but after some hours, suddenly resource usage of istiod pods and kube-apiserver skyrocketed and decreased after around 30 minutes. I have done nothing since I created them but this is repeatedly happening.
I looked into audit logs and `ValidationWebhookConfiguration` keeps getting updated.

Since I am using the 1.17.2 version, whenever the webhook configuration is updated, [this log](https://github.com/istio/istio/blob/1.17.2/pkg/webhooks/validation/controller/controller.go#L440) is supposed to be printed.
But I only found a log that says it will skip the update as it is up-to-date.

```bash
2024-08-19T01:40:27.657058Z	info	validationController	validatingwebhookconfiguration istio-validator-my-revision (failurePolicy=Fail, resourceVersion=6834131698) is up-to-date. No change required.
```

It has been around two days, and its generation is around 40k.

```bash
$ kubectl get validatingwebhookconfiguration istio-validator-my-revision  -o yaml | grep generation
  generation: 41539
```

Is there any other place that updates the `validatingwebhookconfiguration`?
Could anyone guide me on when the `validatingwebhookconfiguration` gets updated?

**Comment by howardjohn** [comment]
No, this is probably something else writing it..? It could be a GitOps control plane like argocd. There was some issues around this that have been fixed since 1.17, which is EOL. I would recommend:
* Check k8s audit logs/managed fields to see who is writing
* See if it occurs on a supported version

**Comment by style95** [comment]
@howardjohn
Thank you for your response. I reviewed the Kubernetes audit logs, and it appears that the source IP addresses are the IPs of the Kubernetes nodes where the istiod pods are running. That's why I initially suspected that istiod was responsible.

Could you please point me to the change you mentioned that has been fixed? Since we're not using ArgoCD, I’d like to examine the details to determine if there's anything relevant to our environment.

Is it correct to assume that, with the recent version, the expected behavior is that the validation webhook configuration is not updated at all?

**Comment by howardjohn** [comment]
it should be updated exactly 1 time

**Comment by style95** [comment]
@howardjohn 
Thanks. We cannot upgrade the version at least for now because of the Kubernetes version.
Could you direct me to the above issues or changes?
I appreciate your help.
